test/bdd/features/did-sidetree.feature

```diff
@@ -227,6 +227,67 @@ Feature:
       When client sends request to "https://orb.domain1.com/sidetree/v1/identifiers" to resolve DID document with canonical did
       Then check success response does NOT contain "newService"
 
+    @did_retrieve_by_version_id
+    Scenario: retrieve document by version ID
+        Given the authorization bearer token for "GET" requests to path "/sidetree/v1/identifiers" is set to "READ_TOKEN"
+        And the authorization bearer token for "POST" requests to path "/sidetree/v1/operations" is set to "ADMIN_TOKEN"
+
+        When client sends request to "https://orb.domain1.com/sidetree/v1/operations" to create DID document
+        Then check success response contains "#interimDID"
+        Then we wait 2 seconds
+
+        When client sends request to "https://orb.domain1.com/sidetree/v1/identifiers" to resolve DID document with interim did
+        Then check success response contains "canonicalId"
+
+        When client sends request to "https://orb.domain1.com/sidetree/v1/identifiers" to resolve DID document with canonical did and version index 0
+        Then check success response contains "createKey"
+
+        When client sends request to "https://orb.domain1.com/sidetree/v1/operations" to add public key with ID "firstKey" to DID document
+        Then check for request success
+        Then we wait 2 seconds
+
+        When client sends request to "https://orb.domain1.com/sidetree/v1/identifiers" to resolve DID document with canonical did
+        Then check success response contains "firstKey"
+
+        When client sends request to "https://orb.domain1.com/sidetree/v1/operations" to add public key with ID "secondKey" to DID document
+        Then check for request success
+        Then we wait 2 seconds
+
+        When client sends request to "https://orb.domain1.com/sidetree/v1/identifiers" to resolve DID document with canonical did
+        Then check success response contains "secondKey"
+
+        When client sends request to "https://orb.domain1.com/sidetree/v1/operations" to add public key with ID "thirdKey" to DID document
+        Then check for request success
+        Then we wait 2 seconds
+
+        When client sends request to "https://orb.domain1.com/sidetree/v1/identifiers" to resolve DID document with canonical did
+        Then check success response contains "thirdKey"
+
+        # start version queries
+        When client sends request to "https://orb.domain1.com/sidetree/v1/identifiers" to resolve DID document with canonical did and version index 0
```

Comment:
What does version index mean?

Reply:
just a way of collecting version ids in BDD test - I will change it as per our conversation

Reply:
Done